includes/sanitizers/class-amp-gallery-block-sanitizer.php

```diff
@@ -73,23 +73,40 @@ class AMP_Gallery_Block_Sanitizer extends AMP_Base_Sanitizer {
 	 * @since 0.2
 	 */
 	public function sanitize() {
-		$nodes     = $this->dom->getElementsByTagName( self::$tag );
-		$num_nodes = $nodes->length;
-		if ( 0 === $num_nodes ) {
-			return;
-		}
+		$xpath = new DOMXPath( $this->dom );
+		$expr  = sprintf(
+			'//ul[ %s ]',
+			implode(
+				' or ',
+				[
+					'( parent::figure[ contains( @class, "wp-block-gallery" ) and @data-amp-carousel = "true" ] )',
```

Comment:
The `contains( @class… )` logic here and the line below should be refactored to make use of `normalize-space()` for example:

```
concat( ' ', normalize-space( @class ), ' ' ), ' wp-block-gallery ' )
```

Reply:
Thanks, @westonruter. Is https://github.com/ampproject/amp-wp/pull/3285/commits/9af22c82b6f7abecc4be3d2c9f4e0668ac72fd39 like what you had in mind?